plumbing/transport/common.go

```diff
@@ -11,32 +11,31 @@
 // Each protocol has its own implementation of
 // `NewGitUploadPackService`, but you should generally not use them
 // directly, use this package's `NewGitUploadPackService` instead.
-package clients
+package transport
 
 import (
 	"fmt"
 
-	"gopkg.in/src-d/go-git.v4/plumbing/client/common"
-	"gopkg.in/src-d/go-git.v4/plumbing/client/http"
-	"gopkg.in/src-d/go-git.v4/plumbing/client/ssh"
+	"gopkg.in/src-d/go-git.v4/plumbing/transport/common"
+	"gopkg.in/src-d/go-git.v4/plumbing/transport/http"
+	"gopkg.in/src-d/go-git.v4/plumbing/transport/ssh"
 )
 
 // Protocols are the protocols supported by default.
-var Protocols = map[string]common.GitUploadPackServiceFactory{
-	"http":  http.NewGitUploadPackService,
-	"https": http.NewGitUploadPackService,
-	"ssh":   ssh.NewGitUploadPackService,
+var Protocols = map[string]common.ClientFactory{
+	"http":  http.NewClient,
+	"https": http.NewClient,
+	"ssh":   ssh.NewClient,
 }
 
 // InstallProtocol adds or modifies an existing protocol.
-func InstallProtocol(scheme string, f common.GitUploadPackServiceFactory) {
+func InstallProtocol(scheme string, f common.ClientFactory) {
 	Protocols[scheme] = f
 }
 
-// NewGitUploadPackService returns the appropriate upload pack service
-// among of the set of known protocols: HTTP, SSH. See `InstallProtocol`
-// to add or modify protocols.
-func NewGitUploadPackService(endpoint common.Endpoint) (common.GitUploadPackService, error) {
+// NewClient returns the appropriate client among of the set of known protocols:
+// HTTP, SSH. See `InstallProtocol` to add or modify protocols.
+func NewClient(endpoint common.Endpoint) (common.Client, error) {
```

Comment:
I don't understand the point behind these changes.

Before: a client package that creates new "services": clients.NewGitUploadPackService()
After: a transport package that creates new "clients": transport.NewClient()

None of these things make much sense, maybe we should explain better the purpose of the package. Also the names of the files don't help: a common.go file, a common directory, with another common.go file inside. The word "common" is meaningless and should be avoided in general. Even if there is a reason to call something common, I doubt there is a reason to name 3 different things like that in the same package.